src/test/java/io/managed/services/test/billing/BillingMetricsTest.java

```diff
@@ -1,22 +1,326 @@
+                oldStorageTotal,
+                expectedIncrease,
+                5.00);
+    }
+
+    @Test(priority = 1, dependsOnMethods = {"invokeDataProduction"}, enabled = true)
+    @SneakyThrows
+    public void testMetricIncomingTrafficIncreased() {
+
+        log.info("test correct incoming traffic metric increase when data are produced");
+        // storage before increasing (value snapshot created even before data were produced)
+        double oldTrafficInTotal = metricToSnapshotMap.get(METRIC_TRAFFIC_IN).getObservedValue();
+
+        // calculation of expected increased value in metric, i.e, conversion of produced bytes (messages * size).
+        double expectedIncrease = this.messageSize * this.messageCount;
+        log.info("expected increase in size: {}", expectedIncrease);
+
+        // waiting for metric to be increased within with 5 range
+        KafkaMgmtMetricsUtils.waitUntilExpectedMetricRange(
+                prometheusWebClient,
+                kafka.getId(),
+                metricToSnapshotMap.get(METRIC_TRAFFIC_IN).getQuery(),
+                oldTrafficInTotal,
+                expectedIncrease,
+                10);
+    }
+
+    @Test(priority = 2, enabled = true)
+    @SneakyThrows
+    public void testTrafficMetricOutIncreased() {
+
+        log.info("test correct outcoming traffic metric increase when data are consumed");
+        // storage before increasing (value snapshot created even before data were produced)
+        double oldTrafficOutTotal = metricToSnapshotMap.get(METRIC_TRAFFIC_OUT).getObservedValue();
+
+        // expected increased value in metric, i.e, conversion of produced bytes (messages * size) * number of consumers  + 15% for handshakes + metatada + extra packet size.
+        double expectedIncrease = this.messageSize * this.messageCount * this.consumerCount * 1.15;
+        log.info("expected increase in size: {}", expectedIncrease);
+
+        // waiting for metric to be increased within with 5 range
+        KafkaMgmtMetricsUtils.waitUntilExpectedMetricRange(
+                prometheusWebClient,
+                kafka.getId(),
+                metricToSnapshotMap.get(METRIC_TRAFFIC_OUT).getQuery(),
+                oldTrafficOutTotal,
+                expectedIncrease,
+                10.0);
+    }
+
+    @SneakyThrows
+    private void createTopics() {
+        var topics = kafkaInstanceApi.getTopics();
+        for (Topic topic : topics.getItems()) {
+            if (topic.getName().equals(TOPIC_NAME)) {
+                return;
+            }
+        }
+
+        // create topic with 3 replicas (the only possible way to create topic currently)
+        kafkaInstanceApi.createTopic(
+                new NewTopicInput()
+                        .name(TOPIC_NAME)
+                        .settings(
+                                new TopicSettings()
+                                        .numPartitions(1)));
```

Comment:
should we set this to 3? If I remember correctly, it will default to 3 anyway right?

Reply:
more partitions only mean that more of .log files will be created in  Strimzi Kafka Pods (which will play role in more storage used than we expect)
The default that we talked about is Replication Factor (how many replicas of data will be created) whereas Partition count says "to how many partition topic (i.e., data) will be split"